shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -428,16 +428,22 @@ - (FlutterCompositor*)createFlutterCompositor {
   // TODO(richardjcai): Add support for creating a FlutterCompositor
   // with a nil _viewController for headless engines.
   // https://github.com/flutter/flutter/issues/71606
-  if (!_viewController) {
+  FlutterViewController* viewController = _viewController;
+  if (!viewController) {
     return nil;
   }
 
   __weak FlutterEngine* weakSelf = self;
 
+  flutter::FlutterCompositor::GetViewCallback get_view_callback =
+      [&viewController](uint64_t view_id) {
```

Comment:
You're capturing a fixed snapshot of `_viewController` here, and I don't see anything in `setViewController` that would update this, so it looks like you're going to end up with stale references any time the view controller for the engine changes.